### ISSUE TITLE
[ISSUE] Issue with resource "databricks_job" resource

### ISSUE THREAD

**Issue by Nestor10** [databricks/terraform-provider-databricks#3632]
### Configuration

```hcl
job_cluster {
    job_cluster_key = "k"
    new_cluster {
      num_workers             = 0
      spark_version           = local.spark_version
      data_security_mode      = "SINGLE_USER"
      single_user_name        = data.databricks_service_principal.self.application_id
      instance_pool_id        = data.databricks_instance_pool.shared["small"].id
      driver_instance_pool_id = data.databricks_instance_pool.shared["small"].id
      node_type_id            = null
      driver_node_type_id     = null
      custom_tags = {
        "ResourceClass" = "SingleNode"
      }
      spark_conf = {
        "spark.databricks.cluster.profile" : "singleNode"
        "spark.databricks.dataLineage.enabled" : "true"
        "spark.databricks.delta.optimizeWrites.enabled" : "true"
        "spark.master" : "local[*,4]"
        "spark.scheduler.mode" : "FIFO"
        "spark.sql.files.ignoreMissingFiles" : "true"
      }
      spark_env_vars = {
        JNAME = "zulu11-ca-amd64"
      }
      aws_attributes {
        instance_profile_arn = xxx
      }
    }
```

### Expected Behavior
single node clusters should build

### Actual Behavior
Error: cannot update job: Cluster validation error: Missing required field: settings.cluster_spec.new_cluster.size

### Steps to Reproduce
1. update to 1.46
### work arounds
1. downgrade to 1.45

### Terraform and provider versions
terraform 1.8.4 and 1.46

### Is it a regression?

1.45 and earlier works

### Important Factoids
could be relevant mac m2 cpu arm terraform target

### Would you like to implement a fix?
I would, but no bandwidth

**Comment by zexoor** [comment]
I've had the same error with deployments to Azure based databricks with terraform version 1.4.6 and databricks provider version 1.46 running from linux based gitlab runners.

**Comment by edwardfeng-db** [comment]
Thanks for reaching out, we are actively working on it and will do a patch release to solve it once the fix is in

**Comment by thaiphv** [comment]
What's going on with this provider?

**Comment by pietern** [comment]
A fix for this issue has been released in v1.47.0.
